lib/helpers/query/cast$expr.js

```diff
@@ -0,0 +1,229 @@
+  }
+
+  _omitUndefined(val);
+
+  return val;
+}
+
+function _omitUndefined(val) {
+  const keys = Object.keys(val);
+  for (const key of keys) {
+    if (val[key] === void 0) {
+      delete val[key];
+    }
+  }
+}
+
+// { $op: <number> }
+function castArithmeticSingle(val) {
+  if (!isLiteral(val)) {
+    return val;
+  }
+
+  try {
+    return castNumber(val);
+  } catch (err) {
+    throw new CastError('Number', val);
+  }
+}
+
+// { $op: [<number>, <number>] }
+function castArithmetic(val) {
+  if (!Array.isArray(val)) {
+    if (!isLiteral(val)) {
+      return val;
+    }
+    try {
+      return castNumber(val);
+    } catch (err) {
+      throw new CastError('Number', val);
+    }
+  }
+
+  return val.map(v => {
+    if (!isLiteral(v)) {
+      return v;
+    }
+    try {
+      return castNumber(v);
+    } catch (err) {
+      throw new CastError('Number', v);
+    }
+  });
+}
+
+// { $op: [expression, expression] }
+function castComparison(val, schema, strictQuery) {
+  if (!Array.isArray(val) || val.length !== 2) {
+    throw new Error('Comparison operator must be an array of length 2');
+  }
+
+  val[0] = _castExpression(val[0], schema, strictQuery);
+  const lhs = val[0];
+
+  if (isLiteral(val[1])) {
+    let path = null;
+    let schematype = null;
+    let caster = null;
+    if (isPath(lhs)) {
+      path = lhs.slice(1);
+      schematype = schema.path(path);
+    } else if (typeof lhs === 'object' && lhs != null) {
+      if (isPath(lhs.$year)) {
+        path = lhs.$year.slice(1) + '.$year';
+        caster = castNumber;
+      } else if (isPath(lhs.$month)) {
+        path = lhs.$month.slice(1) + '.$month';
+        caster = castNumber;
+      } else if (isPath(lhs.$week)) {
+        path = lhs.$week.slice(1) + '.$week';
+        caster = castNumber;
+      } else if (isPath(lhs.$dayOfMonth)) {
+        path = lhs.$dayOfMonth.slice(1) + '.$dayOfMonth';
+        caster = castNumber;
+      } else if (isPath(lhs.$dayOfYear)) {
+        path = lhs.$dayOfMonth.slice(1) + '.$dayOfYear';
+        caster = castNumber;
+      } else if (isPath(lhs.$hour)) {
+        path = lhs.$hour.slice(1) + '.$hour';
+        caster = castNumber;
+      } else if (isPath(lhs.$minute)) {
+        path = lhs.$minute.slice(1) + '.$minute';
+        caster = castNumber;
+      } else if (isPath(lhs.$second)) {
+        path = lhs.$second.slice(1) + '.$second';
+        caster = castNumber;
+      }
+    }
+
+    const is$literal = typeof val[1] === 'object' && val[1] != null && val[1].$literal != null;
+    if (schematype != null) {
+      if (is$literal) {
+        val[1] = { $literal: schematype.cast(val[1].$literal) };
+      } else {
+        val[1] = schematype.cast(val[1]);
+      }
+    } else if (caster != null) {
+      if (is$literal) {
+        try {
+          val[1] = { $literal: caster(val[1].$literal) };
+        } catch (err) {
+          throw new CastError(caster.name.slice(4 /* 'cast'.length */), val[1], path + '.$literal');
```

Comment:
Perhaps using String#replace would show intention more clearly.
```suggestion
          throw new CastError(caster.name.replace('cast', ''), val[1], path + '.$literal');
```

Reply:
Fixed in 29b323c